eng/performance/benchmark_jobs.yml

```diff
@@ -0,0 +1,86 @@
+          - name: BenchViewArguments
+            value: '--benchview-submission-name "$(BenchviewCommitName)" --benchview-machinepool perfsnake --generate-benchview-data --benchview-run-type $(BenchViewRunType)'
+          - name: Creator
+            value: dotnet-performance
+          # for public runs we want to run the benchmarks exactly once, no warmup, no pilot, no overhead
+          - name: BenchmarkDotNetArguments
+            value: '--bdn-arguments="--iterationCount 1 --warmupCount 0 --invocationCount 1 --unrollFactor 1 --strategy ColdStart"'
+          - name: HelixApiAccessToken
+            value: ''
+        - ${{ if and(ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest')) }}:
+          - name: BenchViewRunType
+            value: rolling
+          - name: BenchviewCommitName
+            value: '.NET Performance - $(BenchViewRunType) - $(Build.SourceBranchName) $(Build.SourceVersion)'
+          # for private runs we want to generate data (--generate-benchview-data ) and upload it (--upload-to-benchview-container specifies where)
+          - name: BenchViewArguments
+            value: '--benchview-submission-name "$(BenchviewCommitName)" --benchview-machinepool perfsnake --generate-benchview-data --benchview-run-type $(BenchViewRunType) --upload-to-benchview-container ${_Category}'
+          # for private runs we don't provide any custom BDN arguments (the ones from Program.Main are used)
+          - name: BenchmarkDotNetArguments
+            value: ''
+          - group: DotNet-HelixApi-Access
+          - group: dotnet-benchview
+        workspace:
+          clean: all
+        pool:
+          name: ${{ parameters.pool }}
+        container: ${{ parameters.container }}
+        strategy:
+          matrix:
+            ${{ each framework in parameters.frameworks }}:
+              CoreCLR_${{ framework }}:
+                _Category: coreclr
+                _Framework: ${{ framework }}
+                _BuildConfig: ${{ parameters.architecture }}_CoreCLR_$(_Framework)
+              CoreFX_${{ framework }}:
+                _Category: corefx
+                _Framework: ${{ framework }}
+                _BuildConfig: ${{ parameters.architecture }}_CoreFX_$(_Framework)
+        steps:
+        - checkout: self
+          clean: true
+        - template: /eng/common/templates/steps/send-to-helix.yml
+          parameters:
+            HelixSource: 'pr/dotnet/performance/$(Build.SourceBranch)' # sources must start with pr/, official/, prodcon/, or agent/
+            HelixType: 'test/performance_$(_BuildConfig)/'
+            HelixAccessToken: $(HelixApiAccessToken)
+            HelixTargetQueues: ${{ parameters.queue }}
+            Creator: $(Creator)
+            WorkItemTimeout: 14400 # 4 hours
+            ${{ if eq(parameters.container, '') }}: # container is always empty for Windows, variables['Agent.Os'] seems to be not available here
```

Comment:
Your comment is accurate. I was passing the agentOs to my templates in dotnet-optimization. I wonder if the way you've done this is a little fragile, but I understand why you've done it this way.

Reply:
I wish I knew it before ;) I wasted an hour or two trying to figure out why my `if` does not work ;)

I am going to use your way